File: leetcode/q5_longest_palindrome.py

def check_palindrome(s):
    if s == s[::-1]:
        return True
    return False

def longestPalindrome(s):
    """
    :type s: str
    :rtype: str
    """

    if len(s) <= 1:
        return s

    palindrome = s[0]
    for i in range(0, len(s)-1):
        for j in range(i+2, len(s)+1):
            word = s[i:j]
            if check_palindrome(word):
                # print(word)
                if len(word) > len(palindrome):
                    palindrome = word

    return palindrome


if __name__ == '__main__':
    x = longestPalindrome("gphyvqruxjmwhonjjrgumxjhfyupajxbjgthzdvrdqmdouuukeaxhasdsa")
    print(x)
    print(len(x))

File: leetcode/Q5LongestPalindrome.scala

object Q5LongestPalindrome {
    def longestPalindrome(s: String): String = {
        if(s.length <= 1) return s
    
        var palindrome = s(0).toString
        for(i <- 0 to s.length - 1) {
            for(j <- i + 2 to s.length + 1) {
                val word = s.slice(i, j)
                if(word == word.reverse) {
                    println(word)
                    if(word.length > palindrome.length) palindrome = word
                }
            }
        }
        return palindrome
    }
    
    def main(args: Array[String]) {
    	val p = longestPalindrome("flsuqzhtcahnyickkgtfnlyzwjuiwqiexthpzvcweqzeqpmqwkydhsfipcdrsjkefehhesubkirhalgnevjugfohwnlhbjfewiunlgmomxkafuuokesvfmcnvseixkkzekuinmcbmttzgsqeqbrtlwyqgiquyylaswlgfflrezaxtjobltcnpjsaslyviviosxorjsfncqirsjpkgajkfpoxxmvsyynbbovieoothpjgncfwcvpkvjcmrcuoronrfjcppbisqbzkgpnycqljpjlgeciaqrnqyxzedzkqpqsszovkgtcgxqgkflpmrikksaupukdvkzbltvefitdegnlmzeirotrfeaueqpzppnsjpspgomyezrlxsqlfcjrkglyvzvqakhtvfmeootbtbwfhqucbnuwznigoyatvkocqmbtqghybwrhmyvvuchjpvjckiryvjfxabezchynfxnpqaeampvaapgmvoylyutymdhvhqfmrlmzkhuhupizqiujpwzarnszrexpvgdmtoxvjygjpmiadzdcxtggwamkbwrkeplesupagievwsaaletcuxtpsxmbmeztcylsjxvhzrqizdmgjfyftpzpgxateopwvynljzffszkzzqgofdlwyknqfruhdkvmvrrjpijcjomnrjjubfccaypkpfokohvkqndptciqqiscvmpozlyyrwobeuazsawtimnawquogrohcrnmexiwvjxgwhmtpykqlcfacuadyhaotmmxevqwarppknoxthsmrrknu")
    	println(p)
    }
}
